File: csharp/TetrisGameTests/BoardTest.cs
using hu.klenium.tetris.logic.board;
using hu.klenium.tetris.logic.tetromino;
using hu.klenium.tetris.util;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using TetrisGameTests.Helpers;

namespace TetrisGameTests
{
    [TestClass]
    public class BoardTest
    {
        private Board board = new Board(new Dimension(5, 4));
        [TestMethod] public void NewBoardIsEmpty()
        {
            TestUtil.CheckBoardState(board, new string[]{
                ".....",
                ".....",
                ".....",
                "....."
            });
        }
        [TestMethod] public void AddTetrominoes()
        {
            AddTestData();
            TestUtil.CheckBoardState(board, new string[]{
                ".....",
                "....#",
                ".#.##",
                ".####"
            });
        }
        [TestMethod] public void MovingTetrominoInsideEmptyBoard()
        {
            Tetromino tetromino = new Tetromino(0, board);
            Assert.IsTrue(TestUtil.ControlTetromino(tetromino, "DADDDSAADSDAD"));
        }
        [TestMethod] public void MovingTetrominoOutsideBoardBox()
        {
            Tetromino tetromino = new Tetromino(0, board);
            Assert.IsFalse(TestUtil.ControlTetromino(tetromino, "A"));
            Assert.IsFalse(TestUtil.ControlTetromino(tetromino, "DDDD"));
            Assert.IsFalse(TestUtil.ControlTetromino(tetromino, "SSS"));
        }
        [TestMethod] public void InvalidTetrominoMoveInsideUsedBoard()
        {
            AddTestData();
            Tetromino tetromino = new Tetromino(0, board);
            Assert.IsFalse(TestUtil.ControlTetromino(tetromino, "S"));
            Assert.IsFalse(TestUtil.ControlTetromino(tetromino, "DDD"));
        }
        [TestMethod] public void RemoveOneFullRow()
        {
            AddTestData();
            board.RemoveFullRows(0, 4);
            TestUtil.CheckBoardState(board, new string[]{
                ".....",
                "....#",
                ".#.##",
                ".####"
            });
            Tetromino tetromino = new Tetromino(1, board);
            board.AddTetromino(tetromino);
            TestUtil.CheckBoardState(board, new string[]{
                "#....",
                "#...#",
                "##.##",
                "#####"
            });
            board.RemoveFullRows(0, 4);
            TestUtil.CheckBoardState(board, new string[]{
                ".....",
                "#....",
                "#...#",
                "##.##"
            });
        }
        [TestMethod] public void RemoveMultipleFullRows()
        {
            Tetromino tetromino;
            AddTestData();
            board.RemoveFullRows(0, 4);
            tetromino = new Tetromino(6, board);
            TestUtil.ControlTetromino(tetromino, "DS");
            board.AddTetromino(tetromino);
            tetromino = new Tetromino(1, board);
            board.AddTetromino(tetromino);
            board.RemoveFullRows(0, 4);
            TestUtil.CheckBoardState(board, new string[]{
                ".....",
                ".....",
                ".....",
                "#...."
            });
        }

        public void AddTestData()
        {
            /* .....
               ....#
               .#.##
               .#### */
            Tetromino tetromino;
            tetromino = new Tetromino(3, board);
            TestUtil.ControlTetromino(tetromino, "WWWDSS");
            board.AddTetromino(tetromino);
            tetromino = new Tetromino(6, board);
            TestUtil.ControlTetromino(tetromino, "WWDDWDS");
            board.AddTetromino(tetromino);
        }
    }
}

File: java/src/test/java/BoardTest.java
import helpers.TestUtil;
import hu.klenium.tetris.logic.board.Board;
import hu.klenium.tetris.logic.tetromino.Tetromino;
import hu.klenium.tetris.util.Dimension;
import hu.klenium.tetris.util.Point;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class BoardTest {
    private Board board;
    @BeforeEach public void setUp() {
        board = new Board(new Dimension(5, 4));
    }
    @Test public void newBoardIsEmpty() {
        TestUtil.checkBoardState(board, new String[]{
            ".....",
            ".....",
            ".....",
            "....."
        });
    }
    @Test public void addTetrominoes() {
        addTestData();
        TestUtil.checkBoardState(board, new String[]{
            ".....",
            "....#",
            ".#.##",
            ".####"
        });
    }
    @Test public void movingTetrominoInsideEmptyBoard() {
        Tetromino tetromino = new Tetromino(0, board);
        assertTrue(TestUtil.controlTetromino(tetromino, "DADDDSAADSDAD"));
    }
    @Test public void movingTetrominoOutsideBoardBox() {
        Tetromino tetromino = new Tetromino(0, board);
        assertFalse(board.canAddTetromino(tetromino, new Point(0, -1)));
        assertFalse(TestUtil.controlTetromino(tetromino, "A"));
        assertFalse(TestUtil.controlTetromino(tetromino, "DDDD"));
        assertFalse(TestUtil.controlTetromino(tetromino, "SSS"));
    }
    @Test public void invalidTetrominoMoveInsideUsedBoard() {
        addTestData();
        Tetromino tetromino = new Tetromino(0, board);
        assertFalse(TestUtil.controlTetromino(tetromino, "S"));
        assertFalse(TestUtil.controlTetromino(tetromino, "DDD"));
    }
    @Test public void removeOneFullRow() {
        addTestData();
        board.removeFullRows();
        TestUtil.checkBoardState(board, new String[]{
            ".....",
            "....#",
            ".#.##",
            ".####"
        });
        Tetromino tetromino = new Tetromino(1, board);
        board.addTetromino(tetromino);
        TestUtil.checkBoardState(board, new String[]{
            "#....",
            "#...#",
            "##.##",
            "#####"
        });
        board.removeFullRows();
        TestUtil.checkBoardState(board, new String[]{
            ".....",
            "#....",
            "#...#",
            "##.##"
        });
    }
    @Test public void removeMultipleFullRows() {
        Tetromino tetromino;
        addTestData();
        board.removeFullRows();
        tetromino = new Tetromino(6, board);
        TestUtil.controlTetromino(tetromino, "DS");
        board.addTetromino(tetromino);
        tetromino = new Tetromino(1, board);
        board.addTetromino(tetromino);
        board.removeFullRows();
        TestUtil.checkBoardState(board, new String[]{
            ".....",
            ".....",
            ".....",
            "#...."
        });
    }

    public void addTestData() {
            /* .....
               ....#
               .#.##
               .#### */
        Tetromino tetromino;
        tetromino = new Tetromino(3, board);
        TestUtil.controlTetromino(tetromino, "WWWDSS");
        board.addTetromino(tetromino);
        tetromino = new Tetromino(6, board);
        TestUtil.controlTetromino(tetromino, "WWDDWDS");
        board.addTetromino(tetromino);
    }
}